### ISSUE TITLE
transition: Ensure data integrity between nodes and paths in the DB

### ISSUE THREAD

**Issue by tahini** [chairemobilite/transition#405]
Currently, a path's node IDs are in an array field of the path table. This type of data cannot have a foreign key with the nodes table, which means that if a node gets deleted and is used by a path (which is possible through the delete multiple feature and can always be possible from some obscure code path), then the paths has incoherent data, the calculator cannot load the data from cache, so this prevents all calculations from executing correctly.

We should have a table that links the path_id and node_id, with a sequence in path, with foreign keys and on delete cascade. At least the data will remain coherent and it will be easier to validate if a node is used (instead of querying in an array field). Maybe eventually this table can also store the segment data (to or from the node), to make sure the data of the path is also kept coherent.

**Comment by greenscientist** [comment]
@tahini Do you think we could bump the priority of this ? 
The current data structure with the big blob of json makes it harder to extract the information in trRouting directly.

**Comment by greenscientist** [comment]
(Probably to be combined with https://github.com/chairemobilite/transition/issues/247)

**Comment by tahini** [comment]
Indeed, #247 could help with #761 and #802 as well. And it should make the #801 easier... They could all go together in a major path refactoring. And there's probably a few more of those among the issues. We can add all them to a milestone for early next year
